app/use_cases/update_shopping_cart.rb

```diff
@@ -0,0 +1,36 @@
+class UpdateShoppingCart
+  def run!(current_items:, target_items:)
+    item_types_to_keep = []
+    target_items.each do |i|
+      item_types_to_keep << i.item_type
+    end
+
```

Comment:
you can move these two blocks to another use case `CleanShoppingCart.call(current_items:, :target_items)` where you collect items by types and destroy them